services/graph/pkg/service/v0/utils.go

```diff
@@ -203,23 +202,21 @@ func cs3ReceivedSharesToDriveItems(ctx context.Context,
 				},
 			})
 
-			var errCode errorcode.Error
-			errors.As(errorcode.FromCS3Status(shareStat.GetStatus(), err), &errCode)
-
-			switch {
-			// skip ItemNotFound shares, they might have been deleted in the meantime or orphans.
-			case errCode.GetCode() == errorcode.ItemNotFound:
+			if err := errorcode.FromCS3Status(shareStat.GetStatus(), err); err != nil {
+				logger.Debug().Err(err).
+					Str("shareid", receivedShares[0].GetShare().GetId().GetOpaqueId()).
```

Comment:
This could panic :cry:

Reply:
no, because there is always at least one share for a resource. otherwise we would not know the resource existed. The code above the containing for loop collects all resources that we need to stat from all shares the user has received:
```go
	receivedSharesByResourceID := make(map[string][]*collaboration.ReceivedShare, len(receivedShares))
	for _, receivedShare := range receivedShares {
		if receivedShare == nil {
			continue
		}

		rIDStr := storagespace.FormatResourceID(receivedShare.GetShare().GetResourceId())
		receivedSharesByResourceID[rIDStr] = append(receivedSharesByResourceID[rIDStr], receivedShare)
	}
```
then we range over `receivedSharesByResourceID` which will, as a result, always have at least one share ;-)